tsfresh/utilities/dataframe_functions.py

```diff
@@ -409,27 +408,96 @@ def roll_time_series(df_or_dict, column_id, column_sort, column_kind, rolling_di
     rolling_direction = np.sign(rolling_direction)
 
     grouped_data = df.groupby(grouper)
-    maximum_number_of_timeshifts = maximum_number_of_timeshifts or grouped_data.count().max().max()
+    max_timeshift = max_timeshift or grouped_data.count().max().max()
 
-    if np.isnan(maximum_number_of_timeshifts):
+    if np.isnan(max_timeshift):
         raise ValueError("Somehow the maximum length of your time series is NaN (Does your time series container have "
                          "only one row?). Can not perform rolling.")
 
     if rolling_direction > 0:
-        range_of_shifts = range(maximum_number_of_timeshifts, -1, -1)
+        range_of_shifts = range(max_timeshift, -1, -1)
     else:
-        range_of_shifts = range(-maximum_number_of_timeshifts, 1)
+        range_of_shifts = range(-max_timeshift, 1)
+
+    # Todo: not default for columns_sort to be None
+    if column_sort is None:
+        column_sort = "sort"
+        df[column_sort] = range(df.shape[0])
 
     def roll_out_time_series(time_shift):
         # Shift out only the first "time_shift" rows
         df_temp = grouped_data.shift(time_shift)
-        df_temp[column_id] = "id=" + df[column_id].map(str) + ", shift={}".format(time_shift)
+        df_temp[column_id] = df[column_sort]
         if column_kind:
             df_temp[column_kind] = df[column_kind]
         return df_temp.dropna()
 
-    return pd.concat([roll_out_time_series(time_shift) for time_shift in range_of_shifts],
-                     ignore_index=True)
+    df_shift = pd.concat([roll_out_time_series(time_shift) for time_shift in range_of_shifts], ignore_index=True)
+
+    return df_shift.sort_values(by=[column_id, column_sort])
+
+
+def make_forecasting_frame(x, kind, max_timeshift, rolling_direction):
```

Comment:
Yes, this answers more or less also my question above - I think I was more stuck into the rolling use case. You are right!